Dockerfile

```diff
@@ -1,10 +1,26 @@
-# Goreleaser builds a docker image according to this Dockerfile by copying the sysl binary 
-# to a golang:1.13-alpine image and setting up the entrypoint.
+# Builds a docker image by building the sysl binary
+# on Go 1.13.8 (by default) using the current workspace and then copying the bianry
+# to a image and setting up the entrypoint.
 #
 # The produced image is published to https://hub.docker.com/r/anzbank/sysl
 
-FROM golang:1.13-alpine
+ARG go_ver=1.13.8
+ARG alpine_ver=3.11
 
-COPY sysl /
+FROM golang:${go_ver}-alpine${alpine_ver} as builder
+
+RUN apk --no-cache add make
```

Comment:
I think you're going to need to keep git here. `make build` depends on it for setting the build version info. When I build your image, I get:
```
$ docker run --rm sysl info
Build:
  Version      : --2020-03-05T21:00:17Z
  Git Commit   :
  Date         : 2020-03-05T21:00:17Z
  Go Version   : unspecified
  OS           : unspecified
Runtime:
  GOOS/GOARCH  : linux/amd64
  Go Version   : go1.13.8
```

The version is missing the short sha and branch name, and the git commit is missing the short sha. I wonder if it might be best to pass them in from outside, but for now, let's just get it going with what's there and maybe refactor it later.

`make build` does not set the build Go Version or OS, but that's the way it is. I'll raise a follow-up PR to fix that.